syft/pkg/cataloger/golang/licenses.go

```diff
@@ -112,49 +115,39 @@ func (c *goLicenses) getLicensesFromLocal(moduleName, moduleVersion string) ([]p
 	return c.findLicenses(c.localModCacheResolver, moduleSearchGlob(moduleName, moduleVersion))
 }
 
-func (c *goLicenses) getLicensesFromRemote(moduleName, moduleVersion string) ([]pkg.License, error) {
-	if !c.opts.SearchRemoteLicenses {
-		return nil, nil
-	}
-
-	proxies := remotesForModule(c.opts.Proxies, c.opts.NoProxy, moduleName)
-
-	fsys, err := getModule(proxies, moduleName, moduleVersion)
-	if err != nil {
-		return nil, err
-	}
-
-	dir := moduleDir(moduleName, moduleVersion)
+func (c *goLicenseResolver) getLicensesFromRemote(moduleName, moduleVersion string) ([]goLicense, error) {
+	return c.licenseCache.Resolve(fmt.Sprintf("%s/%s", moduleName, moduleVersion), func() ([]goLicense, error) {
+		proxies := remotesForModule(c.opts.Proxies, c.opts.NoProxy, moduleName)
 
-	// populate the mod cache with the results
-	err = fs.WalkDir(fsys, ".", func(filePath string, d fs.DirEntry, err error) error {
+		fsys, err := getModule(proxies, moduleName, moduleVersion)
 		if err != nil {
-			log.Debug(err)
-			return nil
+			return nil, err
 		}
-		if d.IsDir() {
+
+		var out []goLicense
+		err = fs.WalkDir(fsys, ".", func(filePath string, d fs.DirEntry, _ error) error {
+			if !c.lowerLicenseFileNames.Has(strings.ToLower(d.Name())) {
+				return nil
+			}
+			rdr, err := fsys.Open(filePath)
+			if err != nil {
+				log.Debugf("error opening license file %s: %v", filePath, err)
+				return nil
+			}
+			parsed, err := licenses.Parse(rdr, file.NewLocation(path.Join(moduleDir(moduleName, moduleVersion), filePath)))
```

Comment:
here we can capture a URL that `remoteForProxy` could return, instead of a `file.Location`